Examples/Io/Root/src/RootBFieldWriter.cpp

```diff
@@ -35,23 +35,14 @@ void RootBFieldWriter::run(const Config& config,
   TFile* outputFile =
       TFile::Open(config.fileName.c_str(), config.fileMode.c_str());
   if (outputFile == nullptr) {
-    throw std::ios_base::failure("Could not open '" + config.fileName);
+    throw std::ios_base::failure("Could not open '" + config.fileName + "'");
   }
-  outputFile->cd();
-  TTree* outputTree =
-      new TTree(config.treeName.c_str(), config.treeName.c_str());
+  TTree* outputTree = new TTree(config.treeName.c_str(),
+                                config.treeName.c_str(), 99, outputFile);
```

Comment:
what does mean `99`?

Reply:
That's just the default value of the intervening constructor argument. It's called `splitlevel`. but it doesn't really matter what it means. Unfortunately you can't specify C++ keyword arguments like you can with Python.

The main value of this change is to specify the `outputFile` explicitly, rather than relying on the global "current `TDirectory`".